### PR TITLE
Change Header used for impersonation scopes

### LINKED ISSUE #18374: newImpersonatingRoundTripper needs to use current impersonation headers
right now it is using the old "scopes" header, not the extra header.

/assign enj
/assign simo5

@openshift/sig-security

### PR DESCRIPTION
**Pull request by simo5**
Fixes #18374

### CODE REVIEW

File: pkg/cmd/server/origin/handlers.go [hunk 1]
```diff
@@ -155,10 +154,12 @@ func (c *MasterConfig) versionSkewFilter(handler http.Handler, contextMapper api
 	})
 }
 
+const legacyImpersonateUserScopeHeader = "Impersonate-User-Scope"
```
Comment (deads2k) [inline pkg/cmd/server/origin/handlers.go]: Godoc explaining that older servers respected this.  We switched to the new way in 1q 2017.

File: pkg/oauthserver/client/impersonate.go [hunk 2]
```diff
@@ -27,16 +30,18 @@ func newImpersonatingRoundTripper(user user.Info, delegate http.RoundTripper) ht
 
 func (rt *impersonatingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
```
Comment (deads2k) [inline pkg/oauthserver/client/impersonate.go]: you were going to remove this, right?

Comment (simo5) [reply pkg/oauthserver/client/impersonate.go]: yes

File: pkg/cmd/server/origin/handlers.go (hunk 1 above)
Comment (simo5) [reply pkg/cmd/server/origin/handlers.go]: ok

**Comment by simo5** [comment]
@deads2k the last force push should be final, PTAL

**Comment by simo5** [comment]
Flake #12072

**Comment by simo5** [comment]
Just rebased
/lgtm

**Comment by simo5** [comment]
Flake #18412 
/retest

**Comment by simo5** [comment]
Flake #18136 
/retest

**Pull request merged**